precompiles/erc20/errors.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package erc20
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/crypto"
+	evmtypes "github.com/evmos/evmos/v15/x/evm/types"
+)
+
+var (
+	ErrInsufficientAllowance        = errors.New("ERC20: insufficient allowance")
+	ErrTransferAmountExceedsBalance = errors.New("ERC20: transfer amount exceeds balance")
+)
+
+// BuildExecRevertedErr returns a mocked error that should align with the
+// behavior of the original ERC20 Solidity implementation.
+//
+// FIXME: This is not yet producing the correct reason bytes.
+func BuildExecRevertedErr(reason string) (error, error) {
```

Comment:
can we move this to the integration tests instead?

Reply:
This is not yet being used, but I thought that this would be useful once it's refactored to actually built the RevertError type and then also the ERC-6093 compliant errors, so it's not a test-related util